docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,228 @@
+
+### Implementation
+
+The tracking issue can be found [here](https://github.com/celestiaorg/celestia-node/issues/944).
+It provides a more detailed step-by-step process for how the below described design will be implemented.
+
+### Details
+#### Services Deprecation
+The initial step is to deprecate and remove our services in code and stop relying on services as the notion. Instead, 
+the Module should be used, e.g. HeaderService -> HeaderModule.
+
+* We're organically moving towards the direction of modularized libraries. That is, our `share`, `header` and `state`
+packages are getting shaped as independent sovereign modules which now lack their own API definition. 
+* Consistency. Semantically, modules are closer to what high level Celestia project goals and slogans.
+* Disassociate centralization. Services have always been associated with centralized infrastructure with older 
+monolithic services and newer distributed microservice architectures. 
+
+#### Modules Definition
+
+##### Header
+```go
+type HeaderModule interface {
+	// Head returns the node's local head (tip of the chain of the header store).
+    Head(ctx context.Context) (*header.ExtendedHeader, error)
+	// Get returns the header of the given hash from the node's header store.
+    Get(ctx context.Context, hash tmbytes.HexBytes) (*header.ExtendedHeader, error)
+	// GetByHeight returns the header of the given height from the node's header store.
+	// If the header of the given height is not yet available, the request will hang
+	// until it becomes available in the node's header store.
+    GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error)
+    // GetRangeByHeight returns the given range [from:to) of ExtendedHeaders.
+    GetRangeByHeight(ctx context.Context, from, to uint64) ([]*ExtendedHeader, error)
+    // Subscribe creates long-living Subscription for validated ExtendedHeaders.
+    // Multiple Subscriptions can be created.
+    Subscribe() (Subscription, error)    
+	// SyncState returns the current state of the header Syncer. 
+	SyncState() sync.State
+	// SyncWait blocks until the header Syncer is synced to network head. 
+    SyncWait(ctx context.Context) error
+	// SyncHead provides the Syncer's view of the current network head.
+	SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
+
+##### Shares
+```go
+type SharesModule interface {
+	// GetShare returns the Share from the given data Root at the given row/col coordinates.
+	GetShare(ctx context.Context, dah *Root, row, col int) (Share, error)
+	// GetSharesByNamespace returns all shares of the given nID from the given data Root.
+    GetSharesByNamespace(ctx context.Context, root *Root, nID namespace.ID) ([]Share, error)
+    // SharesAvailable subjectively validates if Shares committed to the given data Root are 
+	// available on the network. 
+    SharesAvailable(ctx context.Context, root *Root) error
```

Comment:
[optional][nit] prefixing this method with an `is` may make it clearer to readers that this method has boolean output.

Proposals: `isEveryShareAvailable`. The 99.9% is a placeholder

```suggestion
	// isEveryShareAvailable checks if the Shares committed to in the given data
	// Root are available on the network. Returns nil if higher than 99.9%
	// confidence the shares are available, otherwise returns an error.
	isEveryShareAvailable(ctx context.Context, root *Root) error
```

Ref: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Update: on second thought the return value of this function is `error` so an `is` prefix may be misleading. If the `error` doesn't need to be propagated to callers, we could return `boolean` for clarity. If the error does need to be propagated, we could return `(boolean, error)`. I don't have strong opinions here.